subsys/mgmt/mcumgr/lib/cmd/fs_mgmt/src/fs_mgmt.c

```diff
@@ -5,30 +5,27 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#include <assert.h>
-#include <zephyr/kernel.h>
 #include <limits.h>
-#include <string.h>
```

Comment:
Not a fan of this, this places an unknown dependency on files *as they are today*, for known includes. I gave up searching for which of these files supposedly includes string.h, but this file uses memcpy, it should directly include string.h, it does not matter if another included file includes string.h or not and if said other file drops needing string.h then this code should not break, but would

Reply:
I do understand.
My reasoning for this is: if the problem happens we will fix it, right now it is very improbable for the problem to happen.
On the other side we waste a lot of CPU and IO on some build servers just to discard contents of included headers, that have to be fully read, before we find the final #endif to discard it.

Reply:
So it is half true, yet the cpp has to parse entire file in case SB does something like:
```
#include <some_file.h>
#undef _SOME_FILE_H_
#include <file_that_includes_some_file.h>
```
for some strange reason.
Also, I doubt strace shows vmem IO.

Reply:
Files definitely need to be self-contained in terms of headers, not doing so is a bug. You're basically relying on someone else to magically include it, and then things like the https://github.com/zephyrproject-rtos/zephyr/pull/50987 catastrophe happen.

Reply:
> Files definitely need to be self-contained in terms of headers

OK valid point... If you mean sources. I'll bring the missing includes at the file levels.
If you mean API headers, I am strongly against including anything that is beyond "need-to-know".

> , not doing so is a bug. You're basically relying on someone else to magically include it

I am relying on the current state until you fix it with #50987.  Then it will start failing compilation and gets fixed.

>  and then things like the #50987 catastrophe happen.

Haven't seen planes fall out of sky due to above, so I would rather call in annoyance.

Reply:
> > Files definitely need to be self-contained in terms of headers
> 
> OK valid point... If you mean sources. I'll bring the missing includes at the file levels. If you mean API headers, I am strongly against including anything that is beyond "need-to-know".

I mean **everything**. If a header is not self-contained, you're pushing the _fix_ to the source file, ie, include whatever random header I need before including me (and to guess which one... read me!). So there should be no exceptions. Not doing so is simply a bad practice. https://google.github.io/styleguide/cppguide.html#Self_contained_Headers

> 
> > , not doing so is a bug. You're basically relying on someone else to magically include it
> 
> I am relying on the current state until you fix it with #50987. Then it will start failing compilation and gets fixed.

#50987 fixes a bug, here: https://github.com/zephyrproject-rtos/zephyr/pull/50987/commits/8ec3a6d1a70681b864005241890f0c47b2884f4b. If your code fails after https://github.com/zephyrproject-rtos/zephyr/pull/50987/commits/8ec3a6d1a70681b864005241890f0c47b2884f4b is applied, it means there's a bug you can fix now.

> 
> > and then things like the #50987 catastrophe happen.
> 
> Haven't seen planes fall out of sky due to above, so I would rather call in annoyance.

Until things like https://github.com/zephyrproject-rtos/zephyr/pull/51028 happen.

Reply:
I give up.